src/Optimizer/Transformer/RewriteAmpUrls.php

```diff
@@ -0,0 +1,375 @@
+                $this->configuration->get(RewriteAmpUrlsConfiguration::GEO_API_URL)
+            );
+        }
+    }
+
+    /**
+     * Check whether a given URL should be preloaded.
+     *
+     * @param string $url Url to check.
+     * @return bool Whether the provided URL should be preloaded.
+     */
+    private function shouldPreload($url)
+    {
+        return substr_compare($url, 'v0.js', -5) === 0
+               ||
+               substr_compare($url, 'v0.css', -6) === 0;
+    }
+
+    /**
+     * Calculate the host string to use.
+     *
+     * @return string Host to use.
+     */
+    private function calculateHost()
+    {
+        $lts = $this->configuration->get(RewriteAmpUrlsConfiguration::LTS);
+        $rtv = $this->configuration->get(RewriteAmpUrlsConfiguration::RTV);
+
+        if ($lts && $rtv) {
+            throw InvalidConfiguration::forMutuallyExclusiveFlags(
+                RewriteAmpUrlsConfiguration::LTS,
+                RewriteAmpUrlsConfiguration::RTV
+            );
+        }
+
+        $ampUrlPrefix      = $this->configuration->get(RewriteAmpUrlsConfiguration::AMP_URL_PREFIX);
+        $ampRuntimeVersion = $this->configuration->get(RewriteAmpUrlsConfiguration::AMP_RUNTIME_VERSION);
+
+        $ampUrlPrefix = rtrim($ampUrlPrefix, '/');
+
+        if ($ampRuntimeVersion && $rtv) {
+            $ampUrlPrefix = RuntimeVersion::appendRuntimeVersion($ampUrlPrefix, $ampRuntimeVersion);
+        } elseif ($lts) {
+            $ampUrlPrefix .= '/lts';
+        }
+
+        return $ampUrlPrefix;
+    }
+
+    /**
+     * Add meta element to the document head.
+     *
+     * @param Document $document Document to add the meta to.
+     * @param string   $name     Name of the meta element.
+     * @param string   $content  Value of the meta element.
+     */
+    private function addMeta(Document $document, $name, $content)
+    {
+        $meta = $document->createElement(Tag::META);
+        $meta->setAttribute(Attribute::NAME, $name);
+        $meta->setAttribute(Attribute::CONTENT, $content);
+        $firstScript = $document->xpath->query('./script', $document->head)->item(0);
+        $document->head->insertBefore($meta, $firstScript);
+    }
+
+    /**
+     * Find the reference node to use for adding elements.
+     *
+     * It fetches the last <link> after the viewport as the reference node, falling back to the viewport if needed.
+     *
+     * @param Document $document Document to find the reference node in.
+     * @return Element|null Reference node to use, or null if none found.
+     */
+    private function findReferenceNode(Document $document)
+    {
+        $referenceNode = $document->viewport;
+
+        while (
+            $referenceNode !== null
+            &&
+            $referenceNode->nextSibling instanceof Element
+            &&
+            $referenceNode->nextSibling->tagName === Tag::LINK
+        ) {
+            $referenceNode = $referenceNode->nextSibling;
+        }
```

Comment:
I thought this would fix the test failures in the plugin, but it's not enough. The only proper fix is to fix #125 and put the `ReorderHead` transformer last again. Everything else is just random juggling at this point anyway.

Reply:
I believe this is fixed now by #130.